src/server/kv.rs

```diff
@@ -286,8 +286,10 @@ impl StoreHandler {
 
 fn extract_region_error<T>(res: &StorageResult<T>) -> Option<RegionError> {
     match *res {
-        Err(StorageError::Txn(TxnError::Engine(EngineError::Request(ref err)))) => {
-            Some(err.to_owned())
+        Err(StorageError::Engine(EngineError::Request(ref e))) |
```

Comment:
@BusyJay @morefreeze @siddontang 
Bad news. The `Error::cause` does not follow the error chain but only looks up for 1 level. Seems insufficient in our case.